build-tools/packages/build-tools/src/repoPolicyCheck/handlers/npmPackages.ts

```diff
@@ -585,8 +586,102 @@ export const handlers: Handler[] = [
             return { resolved: resolved };
         },
     },
+    {
+        name: "npm-package-json-prettier",
+        match,
+        handler: (file) => {
+            let json;
+
+            try {
+                json = JSON.parse(readFile(file));
+            } catch (err) {
+                return "Error parsing JSON file: " + file;
+            }
+
+            const hasPrettierScript = Object.prototype.hasOwnProperty.call(
+                json.scripts,
+                "prettier",
+            );
+            const hasPrettierFixScript = Object.prototype.hasOwnProperty.call(
+                json.scripts,
+                "prettier:fix",
+            );
+            const hasFormatScript = Object.prototype.hasOwnProperty.call(json.scripts, "format");
```

Comment:
I think the error you're seeing (`TypeError: Cannot convert undefined or null to object`) is coming from here, when analyzing a file that does not have a `scripts` section.

<img width="378" alt="image" src="https://user-images.githubusercontent.com/716334/204929158-e49b6ed3-cd7c-4848-9454-6a302972568d.png">

Do we need to do the check with that API for some reason, instead of `json.scripts?["prettier"]`?

Reply:
As @tylerbutler mentioned below, using it for the purpose of lint rule.